### PR TITLE
Add baseline for activity recognition

### PR DESCRIPTION
**Pull request by as6520**
In GitLab by @as6520 on Feb 25, 2021, 11:51

This MR updates the protocol to support running multiple algorithms, one of these can be declared baseline.

Depends on https://gitlab.kitware.com/darpa-sail-on/sailon_tinker_launcher/-/merge_requests/3 and https://gitlab.kitware.com/darpa-sail-on/graph-autoencoder/-/merge_requests/16

### CODE REVIEW

**Review by cfunk1210 — APPROVED** [review]

**Comment by cfunk1210** [comment]
Dismissed my own review since found an issue with adding "detectors" for the there domains.  Since this breaks the other domains, I think we should stage all three here before merging rather than merging a breaking change for some use cases.

**Comment by as6520** [comment]
> Dismissed my own review since found an issue with adding "detectors" for the there domains. Since this breaks the other domains, I think we should stage all three here before merging rather than merging a breaking change for some use cases.

@cfunk1210 can you describe the issue that you are facing?

**Comment by cfunk1210** [comment]
I mentioned this in person today but I was talking about all the other configurations not working since new keywords to the configs have been introduced.  I don't think it's a good idea to push this to master until this has been sorted.

**Comment by cfunk1210** [comment]
do we need to make changes in all adaptors like this for this to work?   Specifically, do we need to change all the adaptors for all domains?  
https://gitlab.kitware.com/darpa-sail-on/graph-autoencoder/-/merge_requests/16/diffs#ed982e77e7ee4c1efa5312d0042e781e5da22fe4_29_29

**Comment by as6520** [comment]
@cfunk1210 I have updated the configs to match with the new convention. To fix the CI tests, we would need to rebase it on #81.

**Comment by cfunk1210** [comment]
Sounds good.  I don't have time to check though

**Comment by as6520** [comment]
Rebasing on #81 to fix the CI pipeline.

**Comment by as6520** [comment]
Depends on #81

**Comment by as6520** [comment]
@nrsyed @waxlamp Could you PTAL at this PR? I have addressed the issues that Chris had raised in the original review. The recent rebase was to fix CI since the sail on the server was updated.

**Review by nrsyed — COMMENTED** [review]
Not sure if I've missed anything or if this is even strictly relevant to this PR, but it looks like the structure of the config JSON files has changed so that the `"detector_config"` field is now nested within the `"detectors"` field. Should lines like, for example [this line](https://github.com/darpa-sail-on/sail-on-client/pull/73/files#diff-c54c7bc9fb065bc15831b17931fe45edc31f4b849230c427218a6d5fa9b73cd1R60), use `self.config["detectors"]["detector_config"]` instead of `self.config["detector_config"]`?

**Comment by as6520** [comment]
Good catch @nrsyed! I had not updated CONDDA to match with the detectors convention. I am done with the update, could you take another look?

**Review by nrsyed — APPROVED** [review]
@as6520 LGTM

File: .github/workflows/ci.yml
Comment (waxlamp) [inline .github/workflows/ci.yml:29]: General note: AFAIK, these URLs are not case sensitive. I routinely clone repositories using all lowercase, because I find anything else to be irritating by default.

**Review by waxlamp — COMMENTED** [review]
I'm not qualified to actually review this PR. However, it looks like Najam has done it, so I would go with his recommendations.

File: .github/workflows/ci.yml
Comment (as6520) [reply .github/workflows/ci.yml:29]: Ah didn't realize that, I'll update it in a separate PR that would address https://github.com/darpa-sail-on/sail-on-client/issues/82

Comment (as6520) [reply .github/workflows/ci.yml:29]: @waxlamp would it be okay if I merge this PR?

Comment (waxlamp) [reply .github/workflows/ci.yml:29]: Yes, that's what I meant when I said to go with Najam's review.

**Pull request merged**
